node/src/accountant/mod.rs

```diff
@@ -1236,38 +1429,150 @@ impl From<&PendingPayableFingerprint> for PendingPayableId {
     }
 }
 
-//TODO the data types should change with GH-497 (including signed => unsigned)
-trait PayableExceedThresholdTools {
+trait PayableThresholdTools {
     fn is_innocent_age(&self, age: u64, limit: u64) -> bool;
-    fn is_innocent_balance(&self, balance: i64, limit: i64) -> bool;
-    fn calculate_payout_threshold(&self, payment_thresholds: PaymentThresholds, x: u64) -> f64;
+    fn is_innocent_balance(&self, balance: u128, limit: u128) -> bool;
+    fn calculate_payout_threshold(&self, payment_thresholds: &PaymentThresholds, x: u64) -> u64;
     as_any_dcl!();
 }
 
 #[derive(Default)]
-struct PayableExceedThresholdToolsReal {}
+struct PayableThresholdToolsReal {}
 
-impl PayableExceedThresholdTools for PayableExceedThresholdToolsReal {
+impl PayableThresholdTools for PayableThresholdToolsReal {
     fn is_innocent_age(&self, age: u64, limit: u64) -> bool {
         age <= limit
     }
 
-    fn is_innocent_balance(&self, balance: i64, limit: i64) -> bool {
+    fn is_innocent_balance(&self, balance: u128, limit: u128) -> bool {
         balance <= limit
     }
 
-    fn calculate_payout_threshold(&self, payment_thresholds: PaymentThresholds, x: u64) -> f64 {
-        let m = -((payment_thresholds.debt_threshold_gwei as f64
-            - payment_thresholds.permanent_debt_allowed_gwei as f64)
-            / (payment_thresholds.threshold_interval_sec as f64
-                - payment_thresholds.maturity_threshold_sec as f64));
-        let b = payment_thresholds.debt_threshold_gwei as f64
-            - m * payment_thresholds.maturity_threshold_sec as f64;
-        m * x as f64 + b
+    fn calculate_payout_threshold(&self, payment_thresholds: &PaymentThresholds, x: u64) -> u64 {
+        ThresholdUtils::calculate_sloped_threshold_by_time(payment_thresholds, x) as u64
     }
     as_any_impl!();
 }
 
+pub struct ThresholdUtils {}
+
+impl ThresholdUtils {
+    pub fn slope(payment_thresholds: &PaymentThresholds, for_wei_computation: bool) -> f64 {
+        if payment_thresholds.debt_threshold_gwei <= payment_thresholds.permanent_debt_allowed_gwei
+        {
+            panic!(
+                "Debt threshold ({}) cannot be smaller than permanent debt allowed ({})",
+                payment_thresholds.debt_threshold_gwei,
+                payment_thresholds.permanent_debt_allowed_gwei
+            )
+        }
+        let slope = -(payment_thresholds.debt_threshold_gwei as f64
+            - payment_thresholds.permanent_debt_allowed_gwei as f64)
+            / Self::convert(payment_thresholds.threshold_interval_sec) as f64;
+        if for_wei_computation {
+            slope * 10_u64.pow(9) as f64
+        } else {
+            slope
+        }
+    }
+
+    fn convert(num: u64) -> i64 {
+        checked_conversion::<u64, i64>(num)
+    }
+
+    fn compute_theoretical_interception_with_y_axis(
+        m: f64,
+        debt_threshold: i64,
+        maturity_threshold_sec: i64,
+    ) -> f64 {
+        let m = m.abs();
```

Comment:
abs() seems wrong here: it makes for V-shaped lines instead of straight ones. Maybe you just mean -?